spec/datadog/core/configuration/option_spec.rb

```diff
@@ -553,30 +553,23 @@
     # | default                    | default             | Option is reset |
     # | rc, default                | rc                  | default         |
     # | rc, default                | default             | rc              |
-    {
-      no_precedence: [],
-      remote_configuration: [Datadog::Core::Configuration::Option::Precedence::REMOTE_CONFIGURATION],
-      programmatic: [Datadog::Core::Configuration::Option::Precedence::PROGRAMMATIC],
-      default: [Datadog::Core::Configuration::Option::Precedence::DEFAULT],
-      remote_and_programmatic: [
-        Datadog::Core::Configuration::Option::Precedence::REMOTE_CONFIGURATION,
-        Datadog::Core::Configuration::Option::Precedence::PROGRAMMATIC
-      ],
-      remote_and_default: [
-        Datadog::Core::Configuration::Option::Precedence::REMOTE_CONFIGURATION,
-        Datadog::Core::Configuration::Option::Precedence::DEFAULT
-      ],
-      programmatic_and_default: [
-        Datadog::Core::Configuration::Option::Precedence::PROGRAMMATIC,
-        Datadog::Core::Configuration::Option::Precedence::DEFAULT
-      ],
-      all: [
-        Datadog::Core::Configuration::Option::Precedence::REMOTE_CONFIGURATION,
-        Datadog::Core::Configuration::Option::Precedence::PROGRAMMATIC,
-        Datadog::Core::Configuration::Option::Precedence::DEFAULT
-      ]
-    }.each do |name, precedences|
-      context "for #{name} set" do
+    #
+    # Coding in tests is not great, but the alternative is to hardcode 2^n combinations
+    # With n the number of values in the precedence list
+    # (with fleet and customer defined stable config, this will go up to 64 combinations)
+    combinations = (0..Datadog::Core::Configuration::Option::Precedence::LIST.length).flat_map do |n|
+      Datadog::Core::Configuration::Option::Precedence::LIST.combination(n).to_a
+    end
```

Comment:
I don't think that you need to code all the possible combinations, you just need the majority, which can use transitive rule to imply that all the rest will work too, and if we found a bug - we can fix it.

The list you replace was doing exactly it.